presto-main/src/main/java/com/facebook/presto/execution/QueryStateMachine.java

```diff
@@ -278,7 +278,7 @@ public QueryInfo getQueryInfo(StageInfo rootStage)
                 }
 
                 PlanFragment plan = stageInfo.getPlan();
-                if (plan != null && plan.getPartitionedSourceNode() instanceof TableScanNode) {
+                if (plan != null && plan.getPartitionedSourceNodes().stream().anyMatch(TableScanNode.class::isInstance)) {
```

Comment:
This stats would be wrong if some of the partitioned source nodes are TS, and some aren't.